From 5304e2de786bcea1e9074c03421b26d03751e1d6 Mon Sep 17 00:00:00 2001
From: Derek Higgins <derekh@redhat.com>
Date: Fri, 17 May 2013 12:16:09 +0100
Subject: [PATCH] Save images in a toci cache file or use if present

FOR DEV ONLY
If you want to
o skip building images in place of a cached version,
o mv first-boot scripts instead of rm

place this file in patches.

Change-Id: Ieb3dbc176de7d7ae8bfefa66e82d2a53c73d2a38
---
 bin/disk-image-create        | 10 ++++++++++
 elements/base/dib-first-boot |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/bin/disk-image-create b/bin/disk-image-create
index d71fa26..4413faa 100755
--- a/bin/disk-image-create
+++ b/bin/disk-image-create
@@ -83,6 +83,13 @@ arg_to_elements "$@"
 
 IMAGE_NAME=${IMAGE_NAME%%\.${IMAGE_TYPE}}
 
+CACHEFILE=$TOCI_CACHE_DIR/$(basename $IMAGE_NAME.$IMAGE_TYPE)
+if [ -e $CACHEFILE ] ; then
+    echo Using $CACHEFILE
+    cp $CACHEFILE $IMAGE_NAME.$IMAGE_TYPE
+    exit 0
+fi
+
 mk_build_dir
 create_base
 run_d extra-data
@@ -127,3 +134,6 @@ finalise_base
 unmount_image
 compress_image
 save_image $IMAGE_NAME.$IMAGE_TYPE
+
+cp $IMAGE_NAME.$IMAGE_TYPE $TOCI_CACHE_DIR/$(basename $IMAGE_NAME.$IMAGE_TYPE)
+
diff --git a/elements/base/dib-first-boot b/elements/base/dib-first-boot
index 1e39860..c13a930 100755
--- a/elements/base/dib-first-boot
+++ b/elements/base/dib-first-boot
@@ -6,6 +6,6 @@ touch /var/log/first-boot.d.log
 chmod 0600 /var/log/first-boot.d.log
 
 /usr/local/bin/dib-run-parts /etc/first-boot.d >> /var/log/first-boot.d.log 2>&1
-rm -fr /etc/first-boot.d
+mv /etc/first-boot.d /etc/first-boot.d_
 # delete itself
 rm $0
-- 
1.8.1.4

